File: 2nd_sem/Pull_Requests/Abhay_singh_AI_01/problem1.c
/*  
    Name - Abhay singh sisoodiya 
    branch - 2nd sem (AI)
    roll no. - 01
    description - The following code is written in c-programming language.
*/
#include<stdio.h>

int main(){
    int n ;
    scanf("%d",&n);
    printf("%d",n);
    if (n==1){
        printf("4 2 1");
    }
    else{
        while(n!=1){
            if (n % 2 == 0){
                n/=2;
                printf(" %d",n);
            }
            else{
                n = 3*n+1;
                printf(" %d",n);
            }
        }
    }
    return 0;
}

File: 2nd_sem/Pull_Requests/Abhay_singh_AI_01/problem1.py
""" 
    Name - Abhay singh sisoodiya 
    branch - 2nd sem (AI)
    roll no. - 01
    description - The following code is written in python.
"""
def func(x):
    print(x,end=' ')
    if (x==1):
        print("4 2 1", end=" ")
    else:
        while(x!=1):
            if x%2==0:
                x/=2
                print(int(x),end=" ")
            else:
                x=3*x+1
                print(int(x), end=" ")
m=int(input())
func(m)
